cmd/lakectl/cmd/local_pull.go

```diff
@@ -0,0 +1,99 @@
+package cmd
+
+import (
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	"github.com/go-openapi/swag"
+	"github.com/spf13/cobra"
+	"github.com/treeverse/lakefs/pkg/api"
+	"github.com/treeverse/lakefs/pkg/diff"
+	"github.com/treeverse/lakefs/pkg/local"
+	"golang.org/x/sync/errgroup"
+)
+
+const (
+	localPullMinArgs = 0
+	localPullMaxArgs = 1
+)
+
+var localPullCmd = &cobra.Command{
+	Use:   "pull [directory]",
+	Short: "Fetch latest changes from lakeFS.",
+	Args:  cobra.RangeArgs(localPullMinArgs, localPullMaxArgs),
+	Run: func(cmd *cobra.Command, args []string) {
+		dir := "."
+		if len(args) == localPullMaxArgs {
+			dir = args[1]
+		}
+		force := Must(cmd.Flags().GetBool("force"))
+		syncFlags := getLocalSyncFlags(cmd)
+		localPath, err := filepath.Abs(dir)
+		if err != nil {
+			DieErr(err)
+		}
+
+		idx, err := local.ReadIndex(localPath)
+		if err != nil {
+			DieErr(err)
+		}
+
+		remote, err := idx.GetCurrentURI()
+		if err != nil {
+			DieErr(err)
+		}
+
+		currentBase := remote.WithRef(idx.AtHead)
+		client := getClient()
+		// make sure no local changes
+		localChng := localDiff(cmd.Context(), client, currentBase, idx.LocalPath())
+		if len(localChng) > 0 && !force {
+			DieFmt("there are %d uncommitted changes. Either commit them first, or use --force to overwrite local changes",
+				len(localChng))
```

Comment:
The word overwrite might be missleading 
```suggestion
		if len(localChng) > 0 && !force {
			DieFmt("there are %d uncommitted changes. Either commit them first or use --force to revert local changes",
				len(localChng))
```